### PR TITLE
Feat: CI can be improved

### PR DESCRIPTION
**Pull request by Rgascoin**
Solution: Splitting GA files for more control, updating actions' versions and add caching for node modules

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 17.x
```
Comment (BjrInt) [inline .github/workflows/build.yaml:24]: The node version, is not the same in the 3 workflows (**17** and **16**). Besides you shouldn't use an odd version number (17) [which are unstable](https://github.com/nodejs/release#release-phases)

**Review by BjrInt — CHANGES_REQUESTED** [review]
You turned a single 58 LOC file into three files that are very redundant. Do you have any insight on the performance gain ?

**Comment by Rgascoin** [comment]
> You turned a single 58 LOC file into three files that are very redundant. Do you have any insight on the performance gain ?

Regardless of having one or more files, for each job, it takes the same configuration steps so it's normal for me to have these similar files.

Having multiple files, allow a more clear code, and also you can configure each action on which branch it can take place.
We didn't define a strategy yet, but we can have Tests on main et dev and activating Build & Lint on each branch.


For perfomance:
The Install dependencies part is arround ~21 seconds with the caching, we were more arround ~30 seconds with the old one.

https://github.com/aleph-im/aleph-sdk-ts/actions/runs/3083172498/jobs/4983776979
https://github.com/aleph-im/aleph-sdk-ts/actions/runs/3090243113/jobs/4998791472

It arround 10seconds saving for each Jobs.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (Rgascoin) [reply .github/workflows/build.yaml]: Fixed: https://github.com/aleph-im/aleph-sdk-ts/pull/49/commits/ddff75a4e3970ccaa4a4f5b4d115b8a5ad7f8c82

**Pull request closed without merging**